src/ts/show.ts

```diff
@@ -360,13 +359,31 @@ document.addEventListener('DOMContentLoaded', () => {
       // loop over it and lock entities
       const results = [];
       checked.forEach(chk => {
-        results.push(EntityC.patchAction(chk.id, Action.Lock));
+        results.push(EntityC.patchAction(chk.id, Action.ForceLock));
       });
 
       Promise.all(results).then(() => {
         reloadEntitiesShow();
       });
 
+    // THE UNLOCK BUTTON FOR CHECKED BOXES
```

Comment:
you should have a "data-what" on the element to distinguish between lock or unlock. Then you don't need to repeat the code and can use one match with data-action `lock-unlock-selected-entities`.

Reply:
thanks. I applied it to the other patch actions as well since the actions were the same, and now they also benefit from the Promise collecting all results before refreshing entities